src/python/pants/backend/native/tasks/cpp_compile.py

```diff
@@ -28,21 +27,18 @@ def implementation_version(cls):
 
   @classmethod
   def subsystem_dependencies(cls):
-    return super(CppCompile, cls).subsystem_dependencies() + (
-      CppCompileSettings.scoped(cls),
-      NativeToolchain.scoped(cls),
-    )
+    return super(CppCompile, cls).subsystem_dependencies() + (CppCompileSettings.scoped(cls),)
 
   @memoized_property
-  def _native_toolchain(self):
-    return NativeToolchain.scoped_instance(self)
-
-  def get_compile_settings(self):
+  def _cpp_compile_settings(self):
     return CppCompileSettings.scoped_instance(self)
 
   @memoized_property
   def _cpp_toolchain(self):
     return self._request_single(LLVMCppToolchain, self._native_toolchain).cpp_toolchain
```

Comment:
That's true. Since we now have testing using both compilers (this was the initial reason for the unconditional selection which nobody else called me out on) this should probably be fixed.

Reply:
I'm thinking of an enum option on NativeBuildSettings or something which is the easiest thing in the entire world to do.

Reply:
Left a comment in the thread -- I have a WIP PR that does exactly that but would depend on this getting merged first. It's not that many lines.